Cargo.toml

```diff
@@ -1,6 +1,6 @@
 [package]
 name = "protoc-grpcio"
-version = "3.0.0"
+version = "4.0.0"
```

Comment:
This may not be necessary, but I bumped the version to be safe since `pub type CompileError = ::failure::Error;` leaks a type that changes before and after this change resulting in a theoretical API breakage, but in practice it shouldn't require much if any refactoring.

Reply:
I think the major bump here is needed for the reasons you pointed surrounding API incompatibility, so, this is the correct move IMO.